### DISCUSSION TITLE
General guidance on creating new containers, and assigning axes

### DISCUSSION THREAD

**Discussion by dan-montelo** [cwapi3d/cwapi3dpython#196] [Q&A]
Hello,

1)  How do you add active elements to a container and set the origin (and axes) of the container?
2)  How do you get the origin and axes of an existing container or container block?  

For **'setting'** containers I have found:
ec.set_container_contents(a0, a1) - but it needs more descriptive documentation
ec.create_auto_container_from_standard(elements, output_name, standard_element_name) - but I don't want to use 'standard' elements
ec.create_auto_container_from_standard_with_reference(elements, output_name, standard_element_name, reference_id) - but I don't want to use 'standard' elements

For **'getting'** from containers I have found:
ec.get_container_content_elements(element_id) - which works.

For **converting** container blocks:
ec.convert_container_to_container_block(elements) - which doesn't seem to work.  

Thanks,

Dan

**Comment by Brunner246** [comment]
Hi @dan-montelo 

You need to work with standard containers. 

1. `create_auto_container_from_standard_with_reference(elements: List[int], output_name: str, standard_element_name: str, reference_id: int) -> int:` 

2. same as for all other elements 
```python
gc.get_xl(container_id)
[1.000000, 0.000000, 0.000000]
gc.get_yl(container_id)
[0.000000, 1.000000, 0.000000]
```

### Set container stub signature

```python
def set_container_contents(container_id: int, element_i_ds: List[int]) -> None:
    """set container contents

    Parameters:
        container_id: container_id
        element_i_ds: element_i_ds

    Returns:
        None
    """ 
 ``` 

### Container to Container Block

Converting from a container to a container block works. 
You have to pass a list of container ids to the function.

```python
ec.convert_container_to_container_block(container_element_ids)
```

**Comment by dan-montelo** [comment]
@Brunner246 

Can you recommend a method to copy Container Blocks?  

When you use: ec.copy_elements([element_id], vector) in the case that the element_id is a Container Block then the returned object is a simple Container.  

Thanks,

Dan
